cmd/malwareExecuteScan.go

```
@@ -118,12 +87,55 @@ func runMalwareScan(config *malwareExecuteScanOptions, telemetryData *telemetry.
 	if scanResponse.MalwareDetected || scanResponse.EncryptedContentDetected {
 		return fmt.Errorf("Malware scan failed for file '%s'. Malware detected: %t, encrypted content detected: %t",
 			config.File, scanResponse.MalwareDetected, scanResponse.EncryptedContentDetected)
-	} else {
-		log.Entry().Infof("Malware scan succeeded for file '%s'. Malware detected: %t, encrypted content detected: %t",
-			config.File, scanResponse.MalwareDetected, scanResponse.EncryptedContentDetected)
+	} 
+	
+	log.Entry().Infof("Malware scan succeeded for file '%s'. Malware detected: %t, encrypted content detected: %t",
+		config.File, scanResponse.MalwareDetected, scanResponse.EncryptedContentDetected)
+
+	return nil
+}
+
+func sendMalwareScanRequest(client piperhttp.Sender, method, url string, candidate io.Reader, header http.Header, cookies []*http.Cookie) (*malwareExecuteScanResponse, error) {
+
+	// piper http utils mashall some http response codes into errors. We wan't to check the status code
+	// ourselvs hence we wait with returning that error (maybe also related to errors others than http status codes)
+
+	response, err := client.SendRequest(method, url, candidate, header, cookies)
+
+	return validateResponse( response, err )
+}
+
+func validateResponse(response *http.Response, err error) (*malwareExecuteScanResponse, error) {
+	
+	var body []byte = []byte("n/a")
+	if response != nil && response.Body != nil {
+		body, _ = ioutil.ReadAll(response.Body)
```

Here we swallow an error caused by reading the response body (`body, _`). Might be helpful for the troubleshooter to get aware of that error rather than having some follow-up failures below.